### ISSUE TITLE
JavaClient: Closing partition-receiver does not free the resources. Re-susbcriptions are failing (0.6.7 and 0.6.9)

### ISSUE THREAD

**Issue by wofr** [Azure/azure-event-hubs#128]
I do have a retry mechanism which re-subscribes me to the event-hub partition in the case the connection went down.
After a while I got the following exception
```
 Failed to create partition receiver for partition 1: com.microsoft.azure.servicebus.ServiceBusException: com.microsoft.azur
e.servicebus.amqp.AmqpException: Exceeded the maximum number of allowed receivers per partition in a consumer group which is 5 TrackingId:6a3faae6-79a1-42cc-8645-0c82234ce705_B45,TimeStamp:4/27/2016 9:14:47 AM TrackingId:ae3ec1e6-50d5-4649-8fa3-19408
61378dd_B45, Reference:c45ecc32-d966-40b7-860f-00e2e5228513, Timestamp:4/27/2016 9:14:48 AM TrackingId:dacb7d3a5d434a609fed0789e3b0dfdb_G32,TimeStamp:4/27/2016 9:14:47 AM, errorContext[NS: data-ingest-services.servicebus.windows.net, PATH: 

```
This telling me basically I’ve reached the max allowed subscription. I thought I did not close the connection correctly that’s why I wrote some test.

Use 5 different clients and subscribe 5 times to the same partition. Four times it works fine, the fives subscription is failing, telling me max 5 is allowed, even so I’ve just subscribed 4 the exception is correct.

Next Test. Use 4 different clients subscribe 4 times to the same partition. Close 4 of the partition-receiver and close 4 of the eventhub client (by using close() and closeSync() ..I tried both). 
Start a new client which subscribes to the same partition ending up in the same exception as above. 

It seems the rescoures are not freed correctly on the AQMP side.
I'm not able to subscribe to this partition again.

**Comment by SreeramGarlapati** [comment]
@wofr: can you pl. provide a code snippet- am looking for pattern like - did you stop all the receives & then issue close? Or issued close and have a receive loop running..?

**Comment by wofr** [comment]
Here we go. Attached my implementation (in scala) my tests and the log of the tests.
Funny enough after the restart of my computer when I executed the test first I could create 5 clients which connected themself to a partition. All test I started afterwards it was just 4.


[EventHubPartitionReceiver(scala).txt](https://github.com/Azure/azure-event-hubs/files/238671/EventHubPartitionReceiver.scala.txt)
[EventHubPartitionReceiverSpec(scala).txt](https://github.com/Azure/azure-event-hubs/files/238672/EventHubPartitionReceiverSpec.scala.txt)
[TestLogOutput.txt](https://github.com/Azure/azure-event-hubs/files/238673/TestLogOutput.txt)

**Comment by SreeramGarlapati** [comment]
Hello @wofr ,

Can you please try this with 0.7.0 - which just released - and confirm this is fixed. Here are 2 minor Breaking changes with 0.7.0:
1.	PartitionReceiveHandler.onClose() API is removed – instead all fatal errors are pumped to onError & the underlying receiver will not close by itself.
2.	PartitionReceiver.receive() API – takes a parameter maxNumberOfEventDataToReceive

Thx!
Sree

**Comment by sjkwak** [comment]
We believe the issue was fixed in recent version (0.7.0). Please reopen the issue if it still happens.
